### PR TITLE
Deliverable B: Review for Coding conventions. GROUP D

### PR DESCRIPTION
**Pull request by revanth-sridhar**
Files assigned to Manoj are: 
- message.cpp (src/data_structures), 
- main.cpp (test/src/sender), 
- sendercadmium.hpp (include/atomics).

Files assigned to Revanth are: 
- receivercadmium.hpp (include/atomics),
- subnetcadmium.hpp (include/atomics),
- message.hpp (include/data_structures).

### CODE REVIEW

File: src/data_structures/message.cpp
Comment (gagandeep8030) [inline src/data_structures/message.cpp:14]: name of the variable "os" could be "out_stream" to be more descriptive.

File: src/data_structures/message.cpp
Comment (gagandeep8030) [inline src/data_structures/message.cpp:23]: Similarly "is" could be replaced with "input_stream".The variable names should be self-explanatory.

File: src/top_model/main.cpp
Comment (gagandeep8030) [inline src/top_model/main.cpp:131]: Missing indentation of 4 spaces.

File: src/top_model/main.cpp
Comment (gagandeep8030) [inline src/top_model/main.cpp:155]: Missing indentation of 4 spaces.

File: src/data_structures/message.cpp
Comment (revanth-sridhar) [reply src/data_structures/message.cpp:14]: Many thanks for this feedback. It is resolved now.

File: src/data_structures/message.cpp
Comment (revanth-sridhar) [reply src/data_structures/message.cpp:23]: Many thanks for this feedback. It is resolved now.

File: src/top_model/main.cpp
Comment (gagandeep8030) [inline src/top_model/main.cpp:134]: Just want to make sure that this line has been arranged according to 80 maximum. characters. There are some more lines in these code too which could have more than 80 characters.

File: src/top_model/main.cpp
Comment (gagandeep8030) [inline src/top_model/main.cpp:130]: The variables names like "outp_1"  could be more descriptive like "out_put1".

File: src/top_model/main.cpp
Comment (gagandeep8030) [inline src/top_model/main.cpp:163]: Maximum length<=80 characters per line

File: src/top_model/main.cpp
Comment (revanth-sridhar) [reply src/top_model/main.cpp:131]: **We did not choose this file for coding conventions**. Please check below files, 

Files assigned to Manoj are:
- message.cpp (src/data_structures),
- main.cpp (test/src/sender),
- sendercadmium.hpp (include/atomics).

Files assigned to Revanth are:
- receivercadmium.hpp (include/atomics),
- subnetcadmium.hpp (include/atomics),
- message.hpp (include/data_structures)

File: src/top_model/main.cpp
Comment (revanth-sridhar) [reply src/top_model/main.cpp:155]: **We did not choose this file for coding conventions**. Please check below files, 

Files assigned to Manoj are:
- message.cpp (src/data_structures),
- main.cpp (test/src/sender),
- sendercadmium.hpp (include/atomics).

Files assigned to Revanth are:
- receivercadmium.hpp (include/atomics),
- subnetcadmium.hpp (include/atomics),
- message.hpp (include/data_structures)

File: src/top_model/main.cpp
Comment (revanth-sridhar) [reply src/top_model/main.cpp:134]: **We did not choose this file for coding conventions**. Please check below files, 

Files assigned to Manoj are:
- message.cpp (src/data_structures),
- main.cpp (test/src/sender),
- sendercadmium.hpp (include/atomics).

Files assigned to Revanth are:
- receivercadmium.hpp (include/atomics),
- subnetcadmium.hpp (include/atomics),
- message.hpp (include/data_structures)

File: src/top_model/main.cpp
Comment (revanth-sridhar) [reply src/top_model/main.cpp:130]: **We did not choose this file for coding conventions**. Please check below files, 

Files assigned to Manoj are:
- message.cpp (src/data_structures),
- main.cpp (test/src/sender),
- sendercadmium.hpp (include/atomics).

Files assigned to Revanth are:
- receivercadmium.hpp (include/atomics),
- subnetcadmium.hpp (include/atomics),
- message.hpp (include/data_structures)

File: src/top_model/main.cpp
Comment (revanth-sridhar) [reply src/top_model/main.cpp:163]: **We did not choose this file for coding conventions**. Please check below files, 

Files assigned to Manoj are:
- message.cpp (src/data_structures),
- main.cpp (test/src/sender),
- sendercadmium.hpp (include/atomics).

Files assigned to Revanth are:
- receivercadmium.hpp (include/atomics),
- subnetcadmium.hpp (include/atomics),
- message.hpp (include/data_structures)

File: include/atomics/senderCadmium.hpp
Comment (gagandeep8030) [inline include/atomics/senderCadmium.hpp:148]: According to coding conventions, in nested loops, open and close brackets should be in the same column.

File: src/top_model/main.cpp
Comment (gagandeep8030) [reply src/top_model/main.cpp:130]: Sorry for the confusion.

File: src/top_model/main.cpp
Comment (gagandeep8030) [reply src/top_model/main.cpp:131]: sorry for the confusion.

File: src/top_model/main.cpp
Comment (gagandeep8030) [reply src/top_model/main.cpp:134]: Sorry for the confusion.

File: src/top_model/main.cpp
Comment (gagandeep8030) [reply src/top_model/main.cpp:155]: Sorry for the confusion.

File: src/top_model/main.cpp
Comment (gagandeep8030) [reply src/top_model/main.cpp:163]: Sorry for the confusion.

File: test/src/sender/main.cpp
Comment (gagandeep8030) [inline test/src/sender/main.cpp]: The variable names ending with "TOP" should be having its ending  with small letters.
For example  iports_TOP=> in_ports_top

File: include/atomics/senderCadmium.hpp
Comment (manoj-goli) [reply include/atomics/senderCadmium.hpp:148]: As we discussed on the last Tuesday, these lines are code has no problem. Therefore, please approve the request.

**Comment by manoj-goli** [comment]
Hi, 

We are waiting for the approval of this pull request. The one comment that been posted has been resolved a couple of days back. Therefore, Kindly review and approve it.

**Review by manoj-goli — APPROVED** [review]

**Comment by gagandeep8030** [comment]
Hi Manoj,

Have you implemented the changes I mentioned for "TOP" in small letters? As I can still see its status as pending.

File: include/atomics/receiverCadmium.hpp
Comment (shubhamthakur463) [inline include/atomics/receiverCadmium.hpp]: As discussed earlier by Gagandeep, the variable names have to be self-descriptive. You can change "os" to something like "output_stream".
Thanks.

File: include/data_structures/message.hpp
Comment (shubhamthakur463) [inline include/data_structures/message.hpp]: Kindly review this portion for indentation(4 Spaces).

File: include/atomics/subnetCadmium.hpp
Comment (shubhamthakur463) [inline include/atomics/subnetCadmium.hpp]: Again, make sure that variable names are self-explanatory.

File: include/data_structures/message.hpp
Comment (revanth-sridhar) [reply include/data_structures/message.hpp]: Many thanks for letting us know on the indentation. We have fixed it now. May I request you to please review them?

File: include/atomics/subnetCadmium.hpp
Comment (revanth-sridhar) [reply include/atomics/subnetCadmium.hpp]: Many thanks for letting us know about the variable names. We have fixed it now. May I request you to please review them ?

File: include/atomics/receiverCadmium.hpp
Comment (revanth-sridhar) [reply include/atomics/receiverCadmium.hpp]: We have changed the variable names in the file. many thanks for letting us know. 
May I request you to please review them again?

File: test/src/sender/main.cpp
Comment (manoj-goli) [reply test/src/sender/main.cpp]: Hi,
Name changes were made as per your comments in the 'respective files that we chose to modify for part B'. Please verify the code and kindly approve it.
Thanks!

**Review by manoj-goli — APPROVED** [review]

**Review by shubhamthakur463 — APPROVED** [review]
This pull has been APPROVED.
Thanks.

**Review by gagandeep8030 — APPROVED** [review]
Hi all I am approving the part B pull request

**Pull request merged**
